arch/risc-v/src/common/riscv_exception.c

```diff
@@ -72,17 +73,26 @@ static const char *g_reasons_str[RISCV_MAX_EXCEPTION + 1] =
 
 int riscv_exception(int mcause, void *regs, void *args)
 {
+  FAR struct tcb_s *tcb = this_task();
   uintptr_t cause = mcause & RISCV_IRQ_MASK;
 
   _alert("EXCEPTION: %s. MCAUSE: %" PRIxREG ", EPC: %" PRIxREG
          ", MTVAL: %" PRIxREG "\n",
          mcause > RISCV_MAX_EXCEPTION ? "Unknown" : g_reasons_str[cause],
          cause, READ_CSR(CSR_EPC), READ_CSR(CSR_TVAL));
 
-  _alert("PANIC!!! Exception = %" PRIxREG "\n", cause);
-  up_irq_save();
-  CURRENT_REGS = regs;
-  PANIC_WITH_REGS("panic", regs);
+  if ((tcb->flags & TCB_FLAG_TTYPE_MASK) != TCB_FLAG_TTYPE_KERNEL)
```

Comment:
Looks good to me and this will work but I'm almost certain they will want you to create multiple versions of "riscv_exception" like "riscv_pagefault" and use those vectors instead. This golden hammer that handles all exceptions is designed to serve as a "better than nothing" solution for the plurality of different exception types.

Reply:
agree, but as long as the multiple versions are not needed, I don't think I'll do those ;)